src/main.js

```diff
@@ -1,10 +1,25 @@
-import { createApp } from "vue";
-import "./style.css";
-import App from "./App.vue";
-import router from "./router";
+import { createApp } from 'vue'
+import './style.css';
+import App from './App.vue'
+import { createPinia } from 'pinia'
+import { useUserStore } from './store'
+import router from './router'
 
-const app = createApp(App);
+async function initializeApp() {
+  const pinia = createPinia();
+  const app = createApp(App);
 
-app.use(router);
+  app.use(pinia);
 
-app.mount("#app");
+  // Load and set the user
+  const store = useUserStore();
+  try {
+    await store.getUser();
+  } catch {}
```

Comment:
We can get the user object before the app initializes so that we can use it to protect routes earlier